### DISCUSSION TITLE
Changing Toast notifications to Hide All when using -RequiredFor AllUsers

### DISCUSSION THREAD

**Discussion by demonjrules** [svrooij/WinTuner#151] [Show and tell]
First off, I love the app. I used to package apps using PowerShell and Chocolatey but only new users would get the latest version but it would never update the apps for people that have had the apps installed for a few months. Keep up the great work!

I could be wrong but I did not see anywhere that allowed you to modify the toast notifications for All Users since I like to push apps silently. Of course, this could be done manually but I am lazy and I may forget in the future.  

I wrote this simple script that utilized the existing token (with Get-WtToken) and then used Graph to modify the toast setting for an app that was already deployed. This only changes to "hide all toast notifications" when the app was deployed with "-RequiredFor AllUsers" but you can certainly rewrite it to match your needs https://learn.microsoft.com/en-us/powershell/module/microsoft.graph.devices.corporatemanagement/set-mgdeviceappmanagementmobileapp?view=graph-powershell-1.0 

```
$AppName = "7-Zip"
$DeployedApp = Get-WtWin32Apps | Where-Object {$_.Name -Eq $AppName}

if($DeployedApp.Count -Eq 0){
	Write-Host "There was no app found with the name $AppName"
} elseif($DeployedApp.Count -GT 1){
	Write-Host "There were multiple apps found with the name $AppName"
	$DeployedApp
} else{
	## we found the 1 app
	Connect-MGGraph -AccessToken $(Get-WtToken | ConvertTo-SecureString -AsPlainText)
	
	$Params = @{
		mobileAppAssignments = @(
			@{
				"@odata.type" = "#microsoft.graph.mobileAppAssignment"
				target = @{
					"@odata.type" = "#microsoft.graph.allLicensedUsersAssignmentTarget"
				}
				intent = "Required"
				settings = @{
					"@odata.type" = "#microsoft.graph.win32LobAppAssignmentSettings"
					notifications = "hideAll"
					installTimeSettings = $Null
					restartSettings = $Null
					deliveryOptimizationPriority = "notConfigured"
				}
			}
		)
	}
	
	Set-MgDeviceAppManagementMobileApp -MobileAppId $DeployedApp.GraphId -BodyParameter $Params
}
```

**Comment by svrooij** [comment]
If I start setting the toast, would there be other “settings” you might want to specify as well?

**Reply by demonjrules** [reply]
The Toast is really all I need but maybe others would want to set something like the restart grace period or installation deadline. I'm not sure if you have seen others mention this before or if I am the only one.
